src/openrct2-ui/interface/Theme.cpp

```diff
@@ -266,45 +271,49 @@ static void ThrowThemeLoadException()
 
 #pragma region UIThemeEntry
 
-json_t* UIThemeWindowEntry::ToJson() const
+json_t UIThemeWindowEntry::ToJson() const
 {
     const WindowThemeDesc* wtDesc = GetWindowThemeDescriptor(WindowClass);
     if (wtDesc == nullptr)
     {
         return nullptr;
     }
 
-    json_t* jsonColours = json_array();
+    json_t jsonColours = json_t::array();
     for (uint8_t i = 0; i < wtDesc->NumColours; i++)
     {
         colour_t colour = Theme.Colours[i];
-        json_array_append_new(jsonColours, json_integer(colour));
+        jsonColours.push_back(colour);
     }
 
-    json_t* jsonEntry = json_object();
-    json_object_set_new(jsonEntry, "colours", jsonColours);
+    json_t jsonEntry = {
+        { "colours", jsonColours },
+    };
 
     return jsonEntry;
 }
 
-UIThemeWindowEntry UIThemeWindowEntry::FromJson(const WindowThemeDesc* wtDesc, const json_t* json)
+UIThemeWindowEntry UIThemeWindowEntry::FromJson(const WindowThemeDesc* wtDesc, json_t& jsonData)
 {
-    json_t* jsonColours = json_object_get(json, "colours");
-    if (jsonColours == nullptr)
+    Guard::Assert(jsonData.is_object(), "UIThemeWindowEntry::FromJson expects parameter jsonData to be object");
+
+    auto jsonColours = Json::AsArray(jsonData["colours"]);
+
+    if (jsonColours.size() == 0)
```

Comment:
```suggestion
    if (jsonColours.empty())
```

Can't we use `.empty()` here?